### PR TITLE
Changed: Documentation about custom AI creation, saving, and uploading

### PR DESCRIPTION
**Pull request by iftwigs**
Changed the tutorials about creating custom Extraction, Document Categorization and File Splitting AIs; added testable code examples and the explanation of how to upload the custom File Splitting AI.

### CODE REVIEW

File: docs/sdk/boilerplates/test_custom_categorization_ai.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    """Test creating and using a custom Categorization AI."""
+    # start init
+    from konfuzio_sdk.trainer.document_categorization import AbstractCategorizationAI
+    from konfuzio_sdk.data import Page
+
+    class CustomCategorizationAI(AbstractCategorizationAI):
+        def __init__(self, *args, **kwargs):
+            pass
+
+        # initialize key variables required by the custom AI
```
Comment (atraining) [inline docs/sdk/boilerplates/test_custom_categorization_ai.py:14]: can we be explicit here?

File: docs/sdk/boilerplates/test_custom_extraction_ai.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    class CustomExtractionAI(AbstractExtractionAI):
+        def __init__(self, *args, **kwargs):
+            pass
+
+        # initialize key variables required by the custom AI
+
+        def fit(self):
+            pass
+
+        # Define architecture and training that the model undergoes, i.e. a NN architecture or a custom hardcoded logic
```
Comment (atraining) [inline docs/sdk/boilerplates/test_custom_extraction_ai.py:19]: what can I do here?

File: konfuzio_sdk/urls.py [hunk 3]
```diff
@@ -204,13 +204,18 @@ def get_annotation_url(document_id: int, annotation_id: int, project_id: int, ho
...
 
-def get_create_ai_model_url(host: str = KONFUZIO_HOST) -> str:
+def get_create_ai_model_url(model_path: str, host: str = KONFUZIO_HOST) -> str:
     """
     Get url to create new AiModel.
 
     :return: URL
     """
-    return f'{host}/api/aimodels/'
+    if 'extraction' in model_path:
```
Comment (zypriafl) [inline konfuzio_sdk/urls.py:213]: I was wondering if we can include the APIV3 in a "smarter" way by generating the api.py file via https://github.com/openapi-generators/openapi-python-client based on the openapi specification (https://app.konfuzio.com/v3/schema/)

Comment (atraining) [reply konfuzio_sdk/urls.py:213]: this idea applies to all apis endpoints and is not the scope of this ticket - can you please add your idea to https://git.konfuzio.com/konfuzio/objectives/-/issues/10937

**Pull request merged**